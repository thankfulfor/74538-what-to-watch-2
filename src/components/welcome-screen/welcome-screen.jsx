import * as React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {history} from '../../history.js';

import AddToFavoritesButton from '../add-to-favorites-button/add-to-favorites-button.jsx';
import FullScreenVideoPlayer from '../fullscreen-video-player/full-screen-video-player.jsx';
import GenreList from '../genre-list/genre-list.jsx';
import Header from '../header/header.jsx';
import {Footer} from '../footer/footer.jsx';
import {PlayButton} from '../play-button/play-button.jsx';

import withOpenCloseButtons from '../../hoc/with-open-close-buttons/with-open-close-buttons.jsx';

import {URLS} from '../../utils/constants.js';

const FullscreenVideoPlayerWithButtonsWrapped = withOpenCloseButtons(FullScreenVideoPlayer, PlayButton);

export const WelcomeScreen = (props) => {
  const {films, promoFilm} = props;

  if (promoFilm === undefined || (Object.entries(promoFilm).length === 0 && promoFilm.constructor === Object)) {
    return null;
  }

  const {
    background_image: backgroundImage,
    background_color: backgroundColor,
    name,
    poster_image: posterImage,
    genre,
    released,
    id
  } = promoFilm;

  return (
    <div>
      <section className="movie-card">
        <div className="movie-card__bg">
          <img src={backgroundImage} alt={name} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <Header />

        <div className="movie-card__wrap">
          <div className="movie-card__info">
            <div className="movie-card__poster">
              <img src={posterImage} alt={name} width="218" height="327" />
            </div>

            <div className="movie-card__desc">
              <h2 className="movie-card__title">{name}</h2>
              <p className="movie-card__meta">
                <span className="movie-card__genre">{genre}</span>
                <span className="movie-card__year">{released}</span>
              </p>

              <div className="movie-card__buttons">

                <FullscreenVideoPlayerWithButtonsWrapped film={promoFilm} />

                <AddToFavoritesButton history={history} filmId={id}/>
              </div>
            </div>
          </div>
        </div>
      </section>

      <div className="page-content" style={{backgroundColor: {backgroundColor}}}>
        <GenreList films={films} />
        <Footer />
      </div>
    </div>
  );
};

WelcomeScreen.propTypes = {
  films: PropTypes.array.isRequired,
  promoFilm: PropTypes.object.isRequired,
  avatarUrl: PropTypes.string.isRequired,
  history: PropTypes.object.isRequired
};

const mapStateToProps = (state, ownProps) => {
  return Object.assign({}, ownProps, {
    avatarUrl: URLS.BASE_URL + state.userData.avatar_url,
    isLoggedIn: state.isLoggedIn,
    films: state.films,
    promoFilm: state.promoFilm
  });
};

export default connect(mapStateToProps)(WelcomeScreen);
