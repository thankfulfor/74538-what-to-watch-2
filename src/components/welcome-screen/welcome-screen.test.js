import React from 'react';
import {WelcomeScreen} from './welcome-screen';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

const films = [{
  picture: `img/we-need-to-talk-about-kevin.jpg`,
  name: `We need to talk about Kevin`,
}];

const promoFilm = {
  picture: `img/we-need-to-talk-about-kevin.jpg`,
  name: `We need to talk about Kevin`,
};

it(`WelcomeScreen корректно рендерится после перезапуска`, () => {
  const clickHandler = function () {};
  const historyMock = {push: jest.fn()};
  const tree = shallow(
      <WelcomeScreen
        history={historyMock}
        avatarUrl={``}
        isLoggedIn={false}
        films={films}
        filteredFilms={films}
        onClick={clickHandler}
        promoFilm={promoFilm}
      />
  );
  expect(toJson(tree)).toMatchSnapshot();
});
