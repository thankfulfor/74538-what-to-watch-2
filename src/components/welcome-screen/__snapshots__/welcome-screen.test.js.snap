// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen корректно рендерится после перезапуска 1`] = `
<div>
  <section
    className="movie-card"
  >
    <div
      className="movie-card__bg"
    >
      <img
        alt="The Grand Budapest Hotel"
      />
    </div>
    <h1
      className="visually-hidden"
    >
      WTW
    </h1>
    <header
      className="page-header movie-card__head"
      onClick={[Function]}
    >
      <div
        className="logo"
      >
        <a
          className="logo__link"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="user-block"
      >
        <div
          className="user-block__avatar"
        >
          <img
            alt="User avatar"
            height="63"
            src="img/avatar.jpg"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      className="movie-card__wrap"
    >
      <div
        className="movie-card__info"
      >
        <div
          className="movie-card__poster"
        >
          <img
            alt="We need to talk about Kevin"
            height="327"
            width="218"
          />
        </div>
        <div
          className="movie-card__desc"
        >
          <h2
            className="movie-card__title"
          >
            We need to talk about Kevin
          </h2>
          <p
            className="movie-card__meta"
          >
            <span
              className="movie-card__genre"
            />
            <span
              className="movie-card__year"
            />
          </p>
          <div
            className="movie-card__buttons"
          >
            <WithOpenCloseButtons />
            <button
              className="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlinkHref="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    className="page-content"
  >
    <Connect(GenreList)
      films={
        Array [
          Object {
            "name": "We need to talk about Kevin",
            "picture": "img/we-need-to-talk-about-kevin.jpg",
          },
        ]
      }
    />
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          © 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
